pkg/deploy/lattice/targets_manager.go

```diff
@@ -61,39 +62,33 @@ func (s *defaultTargetsManager) registerTargets(
 	modelTargets *model.Targets,
 	modelTg *model.TargetGroup,
 ) error {
-	var latticeTargets []*vpclattice.Target
-	for _, modelTarget := range modelTargets.Spec.TargetList {
-		port := modelTarget.Port
-		targetIP := modelTarget.TargetIP
-		t := vpclattice.Target{
-			Id:   &targetIP,
-			Port: &port,
-		}
-		latticeTargets = append(latticeTargets, &t)
-	}
-
 	// No targets to register
-	if len(latticeTargets) == 0 {
+	if len(modelTargets.Spec.TargetList) == 0 {
 		return nil
 	}
+	latticeTargets := modelTargetsToLatticeTargets(modelTargets.Spec.TargetList)
+	// Partition the targets into groups of 100, because that is the allowed max number of targets per RegisterTargets API call
+	// https://docs.aws.amazon.com/vpc-lattice/latest/APIReference/API_RegisterTargets.html
+	partitionedLatticeTargets := getPartitionedLatticeTargets(latticeTargets, 100)
+	var registerTargetsError error
+	for i, targets := range partitionedLatticeTargets {
+		registerRouteInput := vpclattice.RegisterTargetsInput{
+			TargetGroupIdentifier: &modelTg.Status.Id,
+			Targets:               targets,
+		}
+		resp, err := s.cloud.Lattice().RegisterTargetsWithContext(ctx, &registerRouteInput)
+		if err != nil {
+			registerTargetsError = errors.Join(registerTargetsError, fmt.Errorf("failed RegisterTargets %s due to %s", modelTg.Status.Id, err))
+		}
+		if len(resp.Unsuccessful) > 0 {
+			registerTargetsError = errors.Join(registerTargetsError, fmt.Errorf("failed RegisterTargets (Unsuccessful=%d) %s, will retry",
+				len(resp.Unsuccessful), modelTg.Status.Id))
```

Comment:
Should we add each unsuccessful target into error? Dont see harm in it.
Why you mention "will retry" but there is not retry in this code. Also we should retry only on unsuccessful targets. Are we retrying full reconciliation loop in case there is 1 unsuccessful registration?

Reply:
Callers' or this function will do LATTICE_RETRY, I can remove "will retry" in this error message. 


> Also we should retry only on unsuccessful targets. Are we retrying full reconciliation loop in case there is 1 unsuccessful registration?

Current controller don't have a place to store unsuccessful registration(if we have a place to easily store unsuccessful targets we could retry only for unsuccessful for sure). Retry to register all targets is safe, as the vpc lattice registerTargets api is idempotent

Reply:
Please remove "retry", code that actually performs retry should log/wrap err. The full Reconcile loop seems overkill... but out of scope.

Reply:
The current justification for requiring full reconcile loop is that we must go through routes to figure out which target groups are affected by such endpoint changes.
There's room for improvement for sure, but might need a big change for this.